contrib/platform/src/com/sun/jna/platform/win32/Kernel32.java

```diff
@@ -1413,6 +1413,71 @@ boolean CreateProcessW(String lpApplicationName, char[] lpCommandLine,
      */
     boolean GetVersionEx(OSVERSIONINFOEX lpVersionInfo);
 
+    /**
+     * The VerifyVersionInfoW function compares a set of operating system
+     * version requirements to the corresponding values for the currently
+     * running version of the system.This function is subject to manifest-based
+     * behavior.
+     * 
+     * @param lpVersionInformation
+     *            A pointer to an OSVERSIONINFOEX structure containing the
+     *            operating system version requirements to compare. The
+     *            dwTypeMask parameter indicates the members of this structure
+     *            that contain information to compare.
+     * 
+     *            You must set the dwOSVersionInfoSize member of this structure
+     *            to sizeof(OSVERSIONINFOEX). You must also specify valid data
+     *            for the members indicated by dwTypeMask. The function ignores
+     *            structure members for which the corresponding dwTypeMask bit
+     *            is not set.
+     * @param dwTypeMask
+     *            A mask that indicates the members of the OSVERSIONINFOEX
+     *            structure to be tested.
+     * @param dwlConditionMask
+     *            The type of comparison to be used for each lpVersionInfo
+     *            member being compared. To build this value, call the
+     *            VerSetConditionMask function once for each OSVERSIONINFOEX
+     *            member being compared.
+     * @return If the currently running operating system satisfies the specified
+     *         requirements, the return value is a nonzero value.
+     * 
+     *         If the current system does not satisfy the requirements, the
+     *         return value is zero and GetLastError returns
+     *         ERROR_OLD_WIN_VERSION.
+     * 
+     *         If the function fails, the return value is zero and GetLastError
+     *         returns an error code other than ERROR_OLD_WIN_VERSION.
+     */
+    boolean VerifyVersionInfoW(OSVERSIONINFOEX lpVersionInformation, int dwTypeMask, long dwlConditionMask);
```

Comment:
This should be `VerifyVersionInfo` -- but that would not work, because OSVERSIONINFOEX is only correctly bound for unicode?! Crap...

Reply:
I tried `VerifyVersionInfo` but got `UnsatisfiedLinkError: Error looking up function 'VerifyVersionInfo'`. Both the W and A variants worked individually.  The inline functions in the header file use the W version, so that's the one I implemented.  Should I add the A version as well?

Reply:
Er, now I see what you mean. This method should properly use `OSVERSIONINFOEXW` which is not defined (but the `OSVERSIONINFOEX` defined is the `W` version.)  

I had to go look at `git blame` to make sure I wasn't at fault there.

Reply:
So I'm thinking about the correct way to handle this.   One thought was to copy the `OSVERSIONINFOEX` structure to `OSVERSIONINFOEXW` (have one extend the other) and mark the non-W version as deprecated.  But there are many other functions which also call it, so we'd have to change all of them as well.  Even though per the Windows documentation, all these old functions are also deprecated in favor of the Version Helper functions, which currently still refer to this structure... and around in circles we go.